File: 0091. Decode Ways/Solution.c
int numDecodings(char* s)
{
    int a = 0, b = 1, c;
    for (int i = 0; s[i]; ++i) {
        c = 0;
        if (s[i] != '0')
            c = b;
        if (i > 0 && s[i - 1] != '0' && ((s[i - 1] - '0') * 10 + s[i] - '0') <= 26)
            c += a;
        a = b;
        b = c;
    }
    return c;
}

File: 0091. Decode Ways/Solution.py
class Solution:
    def numDecodings(self, s: str) -> int:
        a, b = 0, 1
        for i, c in enumerate(s):
            r = 0
            if c != "0":
                r = b
            if i > 0 and s[i - 1] != "0" and int(s[i - 1 : i + 1]) <= 26:
                r += a
            a, b = b, r
        return r
